ajax/ajax.md

```diff
@@ -16,74 +17,94 @@ About 3 hours
+| Server Response       | responds a corresponding HTML page  |  responds with data (preferably JSON ) |
+| Render |  a complete new page is rendered   | only that part of page changes which has requested |
+| Thread Blocked | true   | false |
+| Data Received by | the browser in form of HTML   | Javascript(callback or Promise) |
 
-What is *XML*?  XML is a type of data interchange format, which means that it's a standard format that computers use to send data to one another. It stands for eXtensible Markup Language. While XML was commonly used when AJAX began, nowadays JSON has taken over as the preferred data format. This is because JSON is native to JavaScript. If your application was using XML, the application would have to take the extra step of converting XML into something that JavaScript could read and manipulate.
+For better understanding follow this [Link](https://github.com/ashishnagpal2498/AjaxVsFullLoadPage)
 
-AJAX came into being because of the XMLHttpRequest API, which was developed by Microsoft in the late 1990s. This API was a way for Microsoft's backend servers to communicate with front end clients, specifically with Internet Explorer. Other browsers and backend servers began to adopt this technique and the XMLHttpRequest object became a web standard over time, and as a general concept, it came to be called AJAX.
 
 ### What does AJAX do?
 
 Let's think about how web applications work. Generally, the steps go like this:
-- The end user types a URL into a browser
-- The browser sends a request to a server designated by the URL
-- The server does some processing, probably queries a database, then responds to the browser by sending back HTML and JavaScript
-- The browser renders that HTML and JavaScript into a webpage
+
 
 Let's say this website that the end user is visiting is an e-commerce site. The end user is browsing a list of products, sees a product they want to buy, and clicks the "Buy" button. This action (clicking the "Buy" button) triggers the request/response steps listed above, this time for the purpose of putting the product in the shopping cart.
 
 Before AJAX, the request/response cycle could only happen with the web page *as a whole*. The end user, by clicking the "Buy" button, would cause the backend server to generate a completely new set of HTML and JavaScript, and the browser would reload and render it as an entirely new page.
 
 AJAX, however, enabled the browser to make requests and only re-render *parts* of a webpage. This was beneficial performance-wise, because smaller strings of HTML, rather than all the HTML for the entire page, was all that was needed to be sent by the server and rendered by the browser. This also made the end user's browsing experience smoother: interactions with the webpage resulted in changes on screen without being interrupted by a reload.
 
-### Threads of Execution
+### Remember
+[Thread of Execution](../javascript/javascript-9.md)
```

Comment:
Make `javascript-9.md` to `javascript-9-async.md`

Reply:
No such file exist with that path.